### PR TITLE
fix(designer): respect existing node dimensions when there is a workflow already loaded by the designer

### PR DESCRIPTION
**Pull request by jaimereyesmx**
For situations where there is an existing workflow, respect the node dimensions so that they are not reset.

### CODE REVIEW

File: libs/designer/src/lib/core/state/workflow/workflowInterfaces.ts [hunk 1]
```diff
@@ -32,5 +32,6 @@ export interface WorkflowState {
   newlyAddedOperations: Record<string, string>;
   runInstance: LogicAppsV2.RunInstanceDefinition | null;
   isDirty: boolean;
+  /** @deprecated this will be removed in a follow up PR */
```
Comment (ek68794998) [inline libs/designer/src/lib/core/state/workflow/workflowInterfaces.ts:35]: Can also add this for `setIsGraphLocked`.

**Comment by ek68794998** [comment]
Looks good to me. Will let Riley review and sign off.

File: libs/designer/src/lib/core/parsers/ParseReduxAction.ts [hunk 2]
```diff
@@ -27,11 +27,17 @@ export const initializeGraphState = createAsyncThunk<
...
+    // For situations where there is an existing workflow, respect the node heights so that
+    // they are not reset
+    const previousChildren = workflow?.graph?.children || [];
+    const deserializedChildren = deserializedWorkflow?.graph?.children || [];
+    for (const node of deserializedChildren) {
+      const previousNode = previousChildren.find((item) => item.id === node.id);
+      if (previousNode?.height) {
+        node.height = previousNode.height;
+      }
     }
```
Comment (rllyy97) [inline libs/designer/src/lib/core/parsers/ParseReduxAction.ts:39]: This does not appear to be taking into account subgraphs / scope nodes, those nodes will have their own children so you will need to make this recursive or else only the top level will be considered

File: libs/designer/src/lib/core/parsers/ParseReduxAction.ts [hunk 3]
```diff
@@ -27,11 +28,10 @@ export const initializeGraphState = createAsyncThunk<
     getConnectionsQuery();
     const { definition, connectionReferences, parameters } = workflowDefinition;
     const deserializedWorkflow = BJSDeserialize(definition, runInstance);
-    // The host can decide whether to make updates to the graph object. In situations like updating the workflow
-    // after save, it makes sense to not update the graph to avoid resetting things like dimensions
-    if (workflow.isGraphLocked && workflow.graph) {
-      deserializedWorkflow.graph = workflow.graph;
-    }
+    // For situations where there is an existing workflow, respect the node dimensions so that they are not reset
+    const previousGraphFlattened = flattenWorkflowNodes(deserializedWorkflow?.graph?.children || []);
```
Comment (jaimereyesmx) [inline libs/designer/src/lib/core/parsers/ParseReduxAction.ts:32]: this should be workflow.graph

**Comment by preetriti1** [comment]
@rllyy97 - why dont we have this issue in our portal? is there scenario different from ours?

**Comment by rllyy97** [comment]
@preetriti1 Their use-case involves returning updated workflow objects to the designer after the initial load, we don't do this in portal currently, we make all our changes and save in-place

**Comment by hartra344** [comment]
@preetriti1 @rllyy97 This is likely an issue we'll start running into when we start integrating CoPilot ourselves.

**Review by hartra344 — APPROVED** [review]

**Pull request merged**